release/src/main/scripts/download_github_actions_artifacts.py

```diff
@@ -0,0 +1,231 @@
+  if reply == "y":
+    return True
+  if reply == "n":
+    return False
+  else:
+    return yes_or_no("Uhhhh... please enter ")
+
+
+def get_build_wheels_workflow_id():
+  url = GH_API_URL_WORKLOW_FMT.format(repo_url=REPO_URL)
+  data = requester(url)
+  return data["id"]
+
+
+def get_last_run(workflow_id):
+  url = GH_API_URL_WORKFLOW_RUNS_FMT.format(
+      repo_url=REPO_URL, workflow_id=workflow_id)
+  event_types = ["push", "pull_request"]
+  runs = []
+  for event in event_types:
+    data = requester(
+        url,
+        params={
+            "event": event, "branch": RELEASE_BRANCH
+        },
+    )
+    runs.extend(data["workflow_runs"])
+
+  filtered_commit_runs = list(
+      filter(lambda w: w.get("head_sha", "") == RELEASE_COMMIT, runs))
+  if not filtered_commit_runs:
+    workflow_web_url = GH_WEB_URL_WORKLOW_RUN_FMT.format(
+        workflow_id=workflow_id)
+    raise Exception(f"No runs for workflow. Verify at {workflow_web_url}")
+
+  sorted_runs = sorted(
+      filtered_commit_runs,
+      key=lambda w: dateutil.parser.parse(w["created_at"]),
+      reverse=True,
+  )
+  last_run = sorted_runs[0]
+  print(
+      f"Found last run. SHA: {RELEASE_COMMIT}, created_at: '{last_run['created_at']}', id: {last_run['id']}"
+  )
+  workflow_web_url = GH_WEB_URL_WORKLOW_RUN_FMT.format(
+      workflow_id=last_run["id"])
+  print(f"Verify at {workflow_web_url}")
+  print(
+      f"Optional upload to GCS will be available at:\n"
+      f"\tgs://beam-wheels-staging/{RELEASE_BRANCH}/{RELEASE_COMMIT}-{workflow_id}/"
+  )
+  return last_run
+
+
+def validate_run(run_data):
+  status = run_data["status"]
+  conclusion = run_data["conclusion"]
+  if status == "completed" and conclusion == "success":
+    return run_data
+
+  url = run_data["url"]
+  workflow_web_url = GH_WEB_URL_WORKLOW_RUN_FMT.format(
+      workflow_id=run_data["id"])
+  print(
+      f"Waiting for Workflow run {run_data['id']} to finish. Check on {workflow_web_url}"
+  )
+  start_time = time.time()
+  last_request = start_time
+  spinner = itertools.cycle(["|", "/", "-", "\\"])
+
+  while True:
+    now = time.time()
+    elapsed_time = time.strftime("%H:%M:%S", time.gmtime(now - start_time))
+    print(
+        f"\r {next(spinner)} Waiting to finish. Elapsed time: {elapsed_time}. "
+        f"Current state: status: `{status}`, conclusion: `{conclusion}`.",
+        end="",
+    )
+
+    time.sleep(0.3)
+    if (now - last_request) > 10:
+      last_request = now
+      run_data = requester(url)
+      status = run_data["status"]
+      conclusion = run_data["conclusion"]
+      if status != "completed":
+        continue
+      elif conclusion == "success":
+        print(
+            f"\rFinished in: {elapsed_time}. "
+            f"Last state: status: `{status}`, conclusion: `{conclusion}`.",
+        )
+        return run_data
+      elif conclusion:
+        print("\r")
+        raise Exception(run_data)
```

Comment:
Add a message here like "run completed unsuccessfully" in addition to the data.

Reply:
Good idea. I extended message. Please take a look ,please.